odoo/osv/expression.py

```diff
@@ -971,7 +971,7 @@ def push_result(sql):
 
             if is_operator(leaf):
                 if leaf == NOT_OPERATOR:
-                    push_result(SQL("(NOT (%s))", pop_result()))
+                    push_result(SQL("((%s) IS NOT TRUE)", pop_result()))
```

Comment:
This is a true semantic change, as it makes the negation of a domain its full opposite (instead of its three-value logic opposite).

| `X` | `NOT X` | `X IS NOT TRUE` |
| --- | --- | --- |
| TRUE | FALSE | FALSE |
| FALSE | TRUE | TRUE |
| NULL | NULL | TRUE |

Mmmh I guess it is fine, since the negation of the domain `[('x', 'in', [1, 2])]` gives something like `x NOT IN (1, 2) OR x IS NULL`, which is fully equivalent to `(x IN (1, 2)) IS NOT TRUE`.